cheroot/test/test_more.py

```diff
@@ -0,0 +1,90 @@
+import os
+import itertools
+import collections
+import pytest
+import requests
+import ssl
+import trustme
+from cheroot.ssl.builtin import BuiltinSSLAdapter
+# need to investigate how to initialize and use the PyOpenSSL adapter
+#from cheroot.ssl.pyopenssl import pyOpenSSLAdapter
+
+@pytest.fixture
+def SSLAdapter(request, root_CA, tmpdir):
+    server_cert = root_CA.issue_server_cert("localhost", "127.0.0.1", "::1")
+
+    server_dir = tmpdir.mkdir("server")
+    server_cert_file = server_dir.join("server.cert")
+    server_cert_file.write(server_cert.cert_chain_pems[0].bytes())
+    server_pkey_file = server_dir.join("server_pkey")
+    server_pkey_file.write(server_cert.private_key_pem.bytes())
+    cert_chain_file = server_dir.join("ca.cert")
+    cert_chain_file.write(root_CA.cert_pem.bytes())
+
+    ssl_adapter = BuiltinSSLAdapter(
+        certificate=str(server_cert_file),
+        private_key=str(server_pkey_file),
+        certificate_chain=str(cert_chain_file),
+    )
+    ssl_adapter.context.verify_mode = request.param
+    return ssl_adapter
+
+
+@pytest.fixture
+def client_cert_files(request, root_CA, tmpdir):
+    """Return path to the parametrized client cert or None if None"""
+    CertKeyPair = collections.namedtuple('CertKeyPair', 'cert,pkey,type')
+    if request.param is None:
+        return CertKeyPair(None, None, None)
+
+    cert_type = request.param
+    cert_type_to_args = {
+        "client":            ["localhost", "127.0.0.1", "::1"],
+        "client_ip":         ["127.0.0.1", "::1"],
+        "client_wildcard":   ["*.localhost"],
+        "client_wrong_host": ["cherrypy.org", "github.com"],
+    }
+
+    client_cert = None
+    if cert_type == "client_wrong_ca":
```

Comment:
There's too much logic in the test case, it must go to separate tests, which should be as simple as possible.